### PR TITLE
Infinite scrolling & bug fixes

### PR DESCRIPTION
**Pull request by Xotl**
## What's the purpose of this PR?
* Fixes Color filter.
* Adds infinite scrolling to improve performance.
* Now the total of cards found is on top instead of bottom.

## How can this be tested?
* Run this branch
* Check that all filters work correctly
* When more than 20 cards of results, scroll down and confirm thath you're able to see all of the cards.

### CODE REVIEW

File: web-page/src/components/CardsContainer.js [hunk 1]
```diff
@@ -1,34 +1,76 @@
-import React, { Component } from 'react';
+import React, { useEffect, useRef, useReducer } from 'react';
```
Comment (metalvegetarianoprogresivo) [inline web-page/src/components/CardsContainer.js]:
```suggestion
import React, {
    useEffect,
    useRef,
    useReducer
} from 'react';
```

File: web-page/src/components/CardsContainer.js [hunk 2]
```diff
@@ -1,34 +1,76 @@
...
 import { connect } from 'react-redux';
 import Card from './Card';
+import InfiniteScroll from 'react-infinite-scroller';
```
Comment (metalvegetarianoprogresivo) [inline web-page/src/components/CardsContainer.js]: Order these imports

File: web-page/src/components/CardsContainer.js [hunk 3]
```diff
@@ -1,34 +1,76 @@
...
+    if (cards.length === 0) {
+        return 'Found 0 cards with that criteria'
+    }
+
+    if (prevCards !== cards) {
+        // This is a hack due to the very poor implementation of the 
+        // infinite scroll plugin. Maybe i should make my own.
+        // 
+        // Let's unmount the InfiniteScroll and wait for the next 
+        // re-render in order to get the pagination reset.
```
Comment (metalvegetarianoprogresivo) [inline web-page/src/components/CardsContainer.js]: This comments use this notation `/***/`

File: web-page/src/components/CardsContainer.js (hunk 2 above)
Comment (Xotl) [reply web-page/src/components/CardsContainer.js]: Alphabetic order of the module or the variable/object imported?

**Comment by Jarv1193** [comment]
Where's the Unit test? 👀

**Comment by Xotl** [comment]
> Where's the Unit test? 👀

There's an issue for that... that you should take 😒

File: web-page/src/components/CardsContainer.js (hunk 2 above)
Comment (metalvegetarianoprogresivo) [reply web-page/src/components/CardsContainer.js]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md

File: web-page/src/components/CardsContainer.js (hunk 3 above)
Comment (Xotl) [reply web-page/src/components/CardsContainer.js]: Done!

File: web-page/src/components/CardsContainer.js (hunk 2 above)
Comment (Xotl) [reply web-page/src/components/CardsContainer.js]: Done!

File: web-page/src/components/CardsContainer.js (hunk 1 above)
Comment (Xotl) [reply web-page/src/components/CardsContainer.js]: Done!

**Pull request merged**